packages/discount-code-generator/src/cli.js

```diff
@@ -0,0 +1,217 @@
+import fs from 'fs'
+import npmlog from 'npmlog'
+import PrettyError from 'pretty-error'
+import yargs from 'yargs'
+import json2csv from 'json2csv'
+import csv from 'csv-parser'
+import flatten, { unflatten } from 'flat'
+
+import discountCodeGenerator from './main'
+import { version } from '../package.json'
+
+process.title = 'discountCodeGenerator'
+
+const args = yargs
+  .usage(`
+Usage: $0 [options]
+Generate multiple discount codes to import to the commercetools platform.`,
+  )
+  .showHelpOnFail(false, 'Use --help to display the CLI options.\n')
+
+  .option('help', {
+    alias: 'h',
+  })
+  .help('help', 'Show help text.')
+
+  .option('version', {
+    alias: 'v',
+    type: 'boolean',
+  })
+  .version('version', 'Show version number.', version)
+
+  .option('quantity', {
+    alias: 'q',
+    describe: 'Quantity of discount codes to generate.',
```

Comment:
I am curious to know how to arrive at those numbers but no hard opinion here, I will prefer a hard limit of 1,000,000

If there are going to be piping this to the discount-code-importer (which most likely is the case) then we probably want to limit it down to reduce the stress on the importer and concurrent API calls

Also, I spoke to martin and he mentioned that there is probably a use-case where a customer wants to generate millions of code 😁